django_project/cplus_api/tasks/sync_default_layers.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os
+import typing
+
+import rasterio
+from datetime import datetime
+from django.utils import timezone
+
+from celery import shared_task
+from django.contrib.auth.models import User
+from storages.backends.s3 import S3Storage
+from django.conf import settings
+from django.core.files.storage import FileSystemStorage
+
+from cplus_api.models import (
+    select_input_layer_storage,
+    InputLayer,
+    COMMON_LAYERS_DIR
+)
+from cplus_api.utils.api_helper import get_layer_type
+
+logger = logging.getLogger(__name__)
+
+
+def process_file(
+        storage: typing.Union[FileSystemStorage, S3Storage],
+        owner: User,
+        component_type: str,
+        file: dict
+):
+    """
+    Function to process a file dictionary into an Input Layer
+    :param storage: Django storage instance
+    :type storage: FileSystemStorage or S3Storage
+
+    :param owner: Owner of the input layer
+    :type owner: User
+
+    :param component_type: Component type of the input layer e.g. ncs_pathway
+    :type component_type: str
+
+    :param file: Dictionary of the file info to be processed
+    :type file: dict
+
+    :return: None
+    :rtype: None
+    """
+    input_layer, created = InputLayer.objects.get_or_create(
+        name=os.path.basename(file['Key']),
+        owner=owner,
+        privacy_type=InputLayer.PrivacyTypes.COMMON,
+        component_type=component_type,
+        defaults={
+            'created_on': timezone.now(),
+            'layer_type': get_layer_type(file['Key'])
+        }
+    )
+
+    # Save layer if the file is modified after input layers last saved OR
+    # if input layer is a new record
+    if file['LastModified'] > input_layer.modified_on or created:
+        media_root = storage.location or settings.MEDIA_ROOT
+        download_path = os.path.join(media_root, file['Key'])
```

Comment:
@zamuzakki, for downloading the layer file to read the metadata, can you use NamedTemporaryFile using 'with' context? in case, there is exception, the temp file will be automatically cleared when exiting the context.

https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile